src/Microsoft.AspNet.Authentication.Google/GoogleAuthenticationHelper.cs

```diff
@@ -1,68 +1,45 @@
-using Microsoft.AspNet.Authentication.OAuth;
 using Newtonsoft.Json.Linq;
 
 namespace Microsoft.AspNet.Authentication.Google
 {
     /// <summary>
-    /// Contains information about the login session as well as the user <see cref="System.Security.Claims.ClaimsIdentity"/>.
+    /// Contains static methods that allow to extract user's information from a <see cref="JObject"/>
+    /// instance retrieved from Google after a successful authentication process.
     /// </summary>
-    public class GoogleAuthenticatedContext : OAuthAuthenticatedContext
+    public static class GoogleAuthenticationHelper
     {
         /// <summary>
-        /// Initializes a new <see cref="GoogleAuthenticatedContext"/>.
-        /// </summary>
-        /// <param name="context">The HTTP environment.</param>
-        /// <param name="user">The JSON-serialized Google user info.</param>
-        /// <param name="tokens">Google OAuth 2.0 access token, refresh token, etc.</param>
-        public GoogleAuthenticatedContext(HttpContext context, OAuthAuthenticationOptions options, JObject user, TokenResponse tokens)
-            : base(context, options, user, tokens)
-        {
-            Id = TryGetValue(user, "id");
-            Name = TryGetValue(user, "displayName");
-            GivenName = TryGetValue(user, "name", "givenName");
-            FamilyName = TryGetValue(user, "name", "familyName");
-            Profile = TryGetValue(user, "url");
-            Email = TryGetFirstValue(user, "emails", "value");
-        }
-
-        /// <summary>
         /// Gets the Google user ID.
         /// </summary>
-        public string Id { get; private set; }
+        public static string GetId(JObject user) => TryGetValue(user, "id");
 
         /// <summary>
         /// Gets the user's name.
         /// </summary>
-        public string Name { get; private set; }
+        public static string GetName(JObject user) => TryGetValue(user, "displayName");
 
         /// <summary>
         /// Gets the user's given name.
         /// </summary>
-        public string GivenName { get; set; }
+        public static string GetGivenName(JObject user) => TryGetValue(user, "name", "givenName");
 
         /// <summary>
         /// Gets the user's family name.
         /// </summary>
-        public string FamilyName { get; set; }
+        public static string GetFamilyName(JObject user) => TryGetValue(user, "name", "familyName");
 
         /// <summary>
         /// Gets the user's profile link.
         /// </summary>
-        public string Profile { get; private set; }
+        public static string GetProfile(JObject user) => TryGetValue(user, "url");
 
         /// <summary>
         /// Gets the user's email.
         /// </summary>
-        public string Email { get; private set; }
+        public static string GetEmail(JObject user) => TryGetFirstValue(user, "emails", "value");
 
         private static string TryGetValue(JObject user, string propertyName)
```

Comment:
IMHO, we should simply use `user.Value<string>("id")`, which basically does the same thing (it returns `null` the item is missing from the JSON payload).